app.js

```diff
@@ -23,9 +23,20 @@ app.post('/delta', bodyParser.json(), async (req, res) => {
   }
   for (const d of pathUpdates) {
     const subjectUri = d.subject.value;
-    const subjectType = WATCH_TYPES.find(t => t.predicateToDoc.uri === d.predicate.value).type;
+    const watchedType = WATCH_TYPES.find(t => t.predicateToObject.uri === d.predicate.value);
+    const subjectType = watchedType.type;
+    const casePredicate = watchedType.casePredicate;
     if (await subjectIsTypeInGraph(subjectUri, GRAPH, [subjectType])) {
-      await syncDocsForSubjectInGraph(d.subject.value, subjectType, GRAPH);
+      await syncObjectsForSubjectInGraph(d.subject.value, subjectType, GRAPH);
+      /* 
+        TODO KAS-3484 the reason for this block is to cleanup field relations found on case but not on the subjects.
+        This is really only useful for publications since they see the list but can't edit.
+        Mistakes (wrong added and removed) made by users will just keep existing unless we remove them here.
+        Only used for type fields. not safe to do for docs because of multiple subjects and legacy data.
+      */
+      if (casePredicate === CASE_FIELD_PREDICATE) {
+        await unsyncObjectsForSubjectInGraph(d.subject.value, subjectType, GRAPH);
+      }
```

Comment:
Just making sure I understand correctly: this will remove any synced predicates/values that are in the `case`, but not in the `subcase`, correct? (because subcase is defined as `CASE_FIELD_PREDICATE`)

What happens when OVRB creates a `case` (niet via MR) and adds governmentAreas (`besluitvorming:beleidsveld`)?
I'm inclined to say these should be kept if there are no subcases, similarly to my remark in the frontend PR. Once there are subcases, we can treat it like any other case.

Reply:
yes, if it doesn't exist on the subcases it shouldn't exist on the case.
for ovrb, we only listen to deltas concerning the model `Procedurestap` with the predicate `besluitvorming:beleidsveld`.
Model `Case` with same predicate `besluitvorming:beleidsveld` should not match anything in the `WATCH_TYPES`
So deltas coming from "not via MR" should never send a matching delta. 
I think anyway (correct me if I'm wrong, maybe I missed something)

Reply:
Perhaps I should make a second constant named `SUBCASE_FIELD_PREDICATE` because it can be confusing that we use the same predicate on 2 resources.
One could argue that the relation on subcase is not in the OSLO model, and should be `ext:beleidsveld`.